build/windows/Makefile.ps1

```diff
@@ -0,0 +1,133 @@
+
+Write-Host("Building Certificate generator code...")
+$currentdir =  $PWD
+if ($false -eq (Test-Path -Path $currentdir)) {
+    Write-Host("Invalid current dir : " + $currentdir + " ") -ForegroundColor Red
+    exit
+}
+
+$builddir = Split-Path -Path $currentdir
+if ($false -eq (Test-Path -Path $builddir)) {
+    Write-Host("Invalid build dir : " + $builddir + " ") -ForegroundColor Red
+    exit
+}
+
+$versionFilePath = Join-Path -Path $builddir -child "version"
+if ($false -eq (Test-Path -Path $versionFilePath)) {
+    Write-Host("Version file path incorret or doesnt exist : " + $versionFilePath + " ") -ForegroundColor Red
+    exit
+}
+
+# read the version info
+foreach($line in Get-Content -Path $versionFilePath) {
+    if ([string]$line.startswith("CONTAINER_BUILDVERSION_") -eq $true) {
+         $parts =  $line.split("=")
+         if ($parts.length -lt 2 ) {
+            Write-Host("Invalid content in version file : " + $versionFilePath + " ") -ForegroundColor Red
+            exit
+         }
+         switch ($parts[0]) {
+            "CONTAINER_BUILDVERSION_MAJOR" { $BuildVersionMajor = $parts[1] }
+            "CONTAINER_BUILDVERSION_MINOR" { $BuildVersionMinor = $parts[1] }
+            "CONTAINER_BUILDVERSION_PATCH" { $BuildVersionPatch = $parts[1] }
+            "CONTAINER_BUILDVERSION_BUILDNR" { $BuildVersionBuildNR = $parts[1] }
+            "CONTAINER_BUILDVERSION_DATE" { $BuildVersionDate = $parts[1] }
+            "CONTAINER_BUILDVERSION_STATUS" { $BuildVersionStatus = $parts[1] }
+            default { Write-Host("This field is not expected in the version file : $line") -ForegroundColor Yellow }
+        }
+    }
+}
+
+if ([string]::IsNullOrEmpty($BuildVersionMajor) -or
+    [string]::IsNullOrEmpty($BuildVersionMinor) -or
+    [string]::IsNullOrEmpty($BuildVersionPatch) -or
+    [string]::IsNullOrEmpty($BuildVersionBuildNR) -or
+    [string]::IsNullOrEmpty($BuildVersionDate) -or
+    [string]::IsNullOrEmpty($BuildVersionStatus)) {
+    Write-Host("Expected version info doesnt exist in this version file : " + $versionFilePath + " ") -ForegroundColor Red
+    exit
+}
+# build version format will be [major].[minior].[patch]-[revision]
+$buildVersionString = $BuildVersionMajor + "." + $BuildVersionMinor + "." + $BuildVersionPatch + "-" + $BuildVersionBuildNR
+$buildVersionDate = $BuildVersionDate
+
+
+$certsrcdir = Join-Path -Path $builddir -ChildPath "windows\installer\certificategenerator"
+if ($false -eq (Test-Path -Path $certsrcdir)) {
+    Write-Host("Invalid certificate generator source dir : " + $certsrcdir + " ") -ForegroundColor Red
+    exit
+}
+Write-Host("set the cerificate generator source code director : " + $certsrcdir + " ...")
+Set-Location -Path $certsrcdir
+
+Write-Host("Adding dotnet packages Newtonsoft.json and BouncyCastle ...")
+dotnet add package Newtonsoft.json
+dotnet add package BouncyCastle
+Write-Host("Successfully added dotnet packages") -ForegroundColor Green
+dotnet build  -f $dotnetcoreframework
```

Comment:
will there be any issues with hardcoding the version?

Reply:
> will there be any issues with hardcoding the version?
> shouldnt be . since we have only have cert generation code which should work with this version.If we add any new .NET code which requires different version then we can update it.